src/project/flowr-analyzer-builder.ts

```diff
@@ -0,0 +1,37 @@
+import type { FlowrConfigOptions } from '../config';
+import { amendConfig, cloneConfig, defaultConfigOptions } from '../config';
+import type { DeepWritable } from 'ts-essentials';
+import type { RParseRequests } from '../r-bridge/retriever';
+import { FlowrAnalyzer } from './flowr-analyzer';
+import { retrieveEngineInstances } from '../engines';
+import type { KnownParser } from '../r-bridge/parser';
+
+export class FlowrAnalyzerBuilder {
+	private flowrConfig:      DeepWritable<FlowrConfigOptions> = cloneConfig(defaultConfigOptions);
+	private readonly request: RParseRequests;
+
+	public amendConfig(func: (config: DeepWritable<FlowrConfigOptions>) => FlowrConfigOptions) : FlowrAnalyzerBuilder {
+		this.flowrConfig = amendConfig(this.flowrConfig, func);
+		return this;
+	}
+
+	public setEngine(engine : 'tree-sitter' | 'r-shell') {
```

Comment:
This should be the `KnownParser` type which... Should be renamed to known engine :p

Reply:
`amendConfig` should accept a function that can return `FlowrConfigOptions | void`.
Disabling the eslint warning is fine